.github/workflows/test_and_release.yml

```
@@ -0,0 +1,98 @@
+name: Test and Release
+on:
+  push:
+    branches: [main, dev]
+  pull_request:
+    branches: [main, dev]
+
+jobs:
+  pytest-and-notebooks:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.9", "3.8", "3.7"]
```

```suggestion
        os: [ubuntu-latest]
        python-version: ["3.9", "3.8", "3.7"]
```
Seems a bit overkill to run on all OS's - let's remove some to save time

---

It runs in parallel? We could get issues with windows; I wouldn't want to remove that (then I would rather remove 3.8), but I would rather just run it on all. If speed is an issue, we could ensure that the test does not need to download a spacy pipeline.

---

Fair - let's go with the whole grid